### PR TITLE
add netcdf_subdataset to gdal_params

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts [hunk 1]
```diff
@@ -62,12 +62,17 @@ export class GdalSourceType extends OperatorType {
                     return {
                         channel: channelDict.channel,
                         datatype: channelDict.datatype,
-                        file_name: channelDict.file_name,
+                        file_name: channelDict.file_name ? channelDict.file_name : undefined,
+                        path: channelDict.path ? channelDict.path : undefined,
                         unit: Unit.fromDict(channelDict.unit),
+                        netcdf_subdataset: channelDict.netcdf_subdataset ? channelDict.netcdf_subdataset : undefined,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:68]: Why is it necessary to have
*if not undefined, add the variable, else undefined*?
What was the problem with the setting before?

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts [hunk 2]
```diff
@@ -62,12 +62,17 @@ export class GdalSourceType extends OperatorType {
...
                     };
                 }) : undefined,
+                file_name: dict.gdal_params.file_name ? dict.gdal_params.file_name : undefined,
+                path: dict.gdal_params.path ? dict.gdal_params.path : undefined,
                 coords: dict.gdal_params.coords,
                 provenance: dict.gdal_params.provenance,
+                netcdf_subdataset: dict.gdal_params.netcdf_subdataset ? dict.gdal_params.netcdf_subdataset : undefined,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:75]: same here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts [hunk 3]
```diff
@@ -158,6 +173,7 @@ export class GdalSourceType extends OperatorType {
             channelConfig: options && options.channelConfig ? options.channelConfig : this.channelConfig,
             sourcename: this.sourcename,
             transform: this.transform,
+            gdal_params: this.gdalParams ? this.gdalParams : undefined,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:176]: same here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts [hunk 4]
```diff
@@ -143,10 +153,15 @@ export class GdalSourceType extends OperatorType {
...
-                        file_name: channelDict.file_name,
+                        file_name: channelDict.file_name ? channelDict.file_name : undefined,
+                        path: channelDict.path ? channelDict.path : undefined,
                         unit: channelDict.unit.toDict(),
+                        netcdf_subdataset: channelDict.netcdf_subdataset ? channelDict.netcdf_subdataset : undefined,
                     };
                 }) : undefined,
+                file_name: this.gdalParams.file_name ? this.gdalParams.file_name : undefined,
+                path: this.gdalParams.path ? this.gdalParams.path : undefined,
+                netcdf_subdataset: this.gdalParams.netcdf_subdataset ? this.gdalParams.netcdf_subdataset : undefined,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:164]: same here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts [hunk 5]
```diff
@@ -122,10 +127,15 @@ export class GdalSourceType extends OperatorType {
...
-                        file_name: channelDict.file_name,
+                        file_name: channelDict.file_name ? channelDict.file_name : undefined,
+                        path: channelDict.path ? channelDict.path : undefined,
                         unit: channelDict.unit.toMappingDict(),
+                        netcdf_subdataset: channelDict.netcdf_subdataset ? channelDict.netcdf_subdataset : undefined,
                     };
                 }) : undefined,
+                file_name: this.gdalParams.file_name ? this.gdalParams.file_name : undefined,
+                path: this.gdalParams.path ? this.gdalParams.path : undefined,
+                netcdf_subdataset: this.gdalParams.netcdf_subdataset ? this.gdalParams.netcdf_subdataset : undefined,
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:138]: same here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 1 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:68]: variable is not required by mapping here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 2 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:75]: variable is not required by mapping here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 3 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:176]: variable is not required by mapping here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 4 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:164]: variable is not required by mapping here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 5 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:138]: variable is not required by mapping here

File: projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts:68]: Yes, but if `channelDict.path` is `undefined`, then `path` is already `undefined` if you say `{…, path: channelDict.path, …}`.

Comment (jdroenner) [reply projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts]: ok

**Pull request merged**